### PR TITLE
Include user id when sending data to Google Analytics

### PR DESCRIPTION
**Pull request by yoomlam**
Connects #15298

### Description
Add code to send the `User.id` (a consecutive auto-generated number referencing a database record) for the logged-in user as a Google Analytics custom metric called `dimension1`.

See https://github.com/department-of-veterans-affairs/caseflow/issues/15298#issuecomment-698661754

`app/views/layouts/application.html.erb:3` 
renders `app/views/layouts/_head.html.erb:16`, 
which renders `app/views/layouts/_head_google_analytics.html.erb:12`

### Testing Plan
1. Open a caseflow page
2. 'View source' and search for 'dimension1'
3. Verify that code does not interfere with current Caseflow functionality

## Confirmation in prod
1. Deploy to prod; ensuring nothing is broken
2. Check Google Analytics for `db_user_id` custom metric and create a report: https://moz.com/blog/custom-dimensions-google-analytics

### CODE REVIEW

File: app/views/layouts/_head_google_analytics.html.erb [hunk 1]
```diff
@@ -9,6 +9,7 @@
       userId: '<%= current_user ? current_user.id : "" %>'
```
Comment (yoomlam) [inline app/views/layouts/_head_google_analytics.html.erb:9]: We're already sending the user id as `userId`, probably to leverage GA's [built-in User-ID capability](https://support.google.com/analytics/answer/3123662?hl=en&ref_topic=3123660). However, it is not currently enabled in GA.
While enabling this User-ID capability could be helpful for some metrics, we cannot create custom segments or reports desired by @lund5000.  As a result, we're using a custom dimension.

Comment (jcq) [inline app/views/layouts/_head_google_analytics.html.erb:9]: Not a huge deal (and not necessarily more legible), but a more succinct syntax could use both optional chaining and nullish coalescing operators, like so:
```js
current_user?.id ?? ""
```

File: app/views/layouts/_head_google_analytics.html.erb [hunk 2]
```diff
@@ -9,6 +9,7 @@
...
     });
     ga('set', 'anonymizeIp', true);
+    ga('set', 'dimension1', <%= current_user ? current_user.id : -1 %>);
```
Comment (jcq) [inline app/views/layouts/_head_google_analytics.html.erb:12]: why are we calling this "dimension1" instead of something more descriptive?

**Review by jcq — APPROVED** [review]
Added a couple of comments, but seems fine in concept. I didn't log into Analytics to check, but it [looks like](https://support.google.com/analytics/answer/2709828?hl=en#configuration) you'll need to ensure that the dimension is defined ahead of time.

File: app/views/layouts/_head_google_analytics.html.erb (hunk 2 above)
Comment (yoomlam) [reply app/views/layouts/_head_google_analytics.html.erb:12]: See https://github.com/department-of-veterans-affairs/caseflow/issues/15298#issuecomment-698661754. It appears that GA maps 'dimension1' to the name that I configured on GA's web UI.

**Comment by yoomlam** [comment]
> Added a couple of comments, but seems fine in concept. I didn't log into Analytics to check, but it [looks like](https://support.google.com/analytics/answer/2709828?hl=en#configuration) you'll need to ensure that the dimension is defined ahead of time.

I created the custom dimension, as mentioned in https://github.com/department-of-veterans-affairs/caseflow/issues/15298#issuecomment-698661754.

File: app/views/layouts/_head_google_analytics.html.erb (hunk 1 above)
Comment (yoomlam) [reply app/views/layouts/_head_google_analytics.html.erb:9]: Used `current_user&.id || ""` since it's expecting Ruby.

**Pull request merged**